composer/models/huggingface.py

```diff
@@ -544,26 +543,19 @@ def eval_forward(self, batch, outputs: Optional[Any] = None):
                 ]
 
         if self.use_logits or batch.get('mode', None) == 'icl_task':
-            # pop labels first to avoid computing loss
```

Comment:
Instead of `outputs` I took `metric` instead but overall this should be the same. Due to the variable nature of `labels` I used `batch_to_device` to handle multiple cases at once